gradio/cli/commands/components/publish.py

```diff
@@ -219,10 +223,22 @@ def _publish(
             source_dir_ = source_dir_ or str(Path("."))
             source_dir = Path(source_dir_).resolve()
     if upload_demo:
+        package_name, version = wheel_file.name.split("-")[:2]
```

Comment:
might be better to pull these from the pyproject toml

Reply:
Yeah, I was thinking that but we don't actually _require_ the source directory here. Should we just assume it is `./pyproject.yaml` if none is passed?

Reply:
`.toml` even